src/libguac/guacamole/protocol-types.h

```diff
@@ -276,5 +278,50 @@ typedef enum guac_line_join_style {
     GUAC_LINE_JOIN_ROUND = 0x2
 } guac_line_join_style;
 
+/**
+ * Track the various protocol versions supported by guacd to help negotiate
+ * features that may not be supported by various versions of the client.
+ */
+typedef enum guac_protocol_version {
+    /**
+     * Original protocol version 1.0.0, which lacks support for negotiating
+     * parameters and protocol version.
+     */
+    VERSION_1_0_0 = 100,
+            
+    /**
+     * Protocol version 1.1.0, which includes support for parameter and version
+     * negotiation and for sending timezone information from the client
+     * to the server.
+     */
+    VERSION_1_1_0 = 110,
+            
+    /**
+     * Protocol version 1.2.0, which supports the "required" instruction,
+     * allowing connections in guacd to request information from the client and
+     * await a response.
+     */
+    VERSION_1_2_0 = 120
+
+} guac_protocol_version;
```

Comment:
With 1.2.0 now being released, and the new `required` instruction being part of 1.3.0, we'll need a constant for that version, as well.

Reply:
So, I guess my question, here, is: is there actually a protocol version 1.2.0?  Since, protocol-wise, 1.2.0 is identical to 1.1.0, and we're making the changes in 1.3.0?

Reply:
Ah, I see what you're saying. Yes, that's correct, there would be no protocol version 1.2.0.